### PR TITLE
feat: Add Emoji Autocomplete functionality that displays matching emoji suggestions on : colon type

### LINKED ISSUE #1013: Feature: Add Emoji Autocomplete on Colon Type
As an EmbeddedChat developer

**I need to** see a list of suggested emojis when I type a colon `:` followed by text in the message box.

**So That** I can quickly find and insert emojis into my message without having to remember their exact names.

**Acceptance Criteria**

- [ ] When a user types a colon (:) followed by at least two characters (e.g., :la), a suggestion pop-up appears above the input box.

- [ ] The pop-up displays a scrollable list of emojis that match the text typed after the colon.

This feature is already available in the main Rocket.Chat web and desktop clients, providing a faster workflow for keyboard-focused users.

Image

### PR DESCRIPTION
**Pull request by snehalsaurabh**
# Add Emoji Autocomplete on Colon Type

## Acceptance Criteria fulfillment

- [x] When a user types a colon (:) followed by at least two characters (e.g., :la), a suggestion pop-up appears above the input box.
- [x] The pop-up displays a scrollable list of emojis that match the text typed after the colon.
- [x] Feature provides faster workflow for keyboard-focused users, matching main Rocket.Chat web and desktop clients.

Fixes #1013 (issue)

## Video/Screenshots
Demo Video: [EmbeddedChat - Emoji Autocomplete Feature](https://www.loom.com/share/ad88ba5a481b4ada8be4a30aa2b32740?sid=caf95061-948c-4ca0-824c-9ff32f9b6aa9)
<br>
Screenshot 2025-10-09 063910

## Key Features Implemented
✅ Trigger: : + 2 characters minimum
✅ Display format: emoji unicode + :shortname: (matching Rocket.Chat)
✅ Keyboard navigation: ArrowUp/Down, Enter, Escape
✅ Click selection support
✅ Auto-scroll to highlighted item
✅ Consistent styling with existing autocomplete lists
✅ Limited to 10 results for better performance
✅ Proper state management and cleanup
✅ No interference with @ mentions or / commands

## Implementation Details
This PR adds emoji autocomplete functionality to the EmbeddedChat component, allowing users to quickly find and insert emojis by typing : followed by text in the message input box. The implementation follows the same patterns as existing autocomplete features (mentions and commands) for consistency.

#### New Files:
- `packages/react/src/lib/emojiList.js` - Comprehensive emoji data source with shortnames and aliases
- `packages/react/src/hooks/useSearchEmoji.js` - Hook for emoji filtering logic
- `packages/react/src/views/EmojiList/` - Complete EmojiList component with styling

#### Modified Files:
- `packages/react/src/views/ChatInput/ChatInput.js` - Integrated emoji autocomplete functionality

### Technical Highlights:
- Performance Optimized: Limited to 10 suggestions with efficient filtering
- Accessibility: Full keyboard navigation support
- Theme Integration: Consistent styling with existing UI components
- State Management: Proper cleanup and no memory leaks
- User Experience: Smooth interaction with existing features

## PR Test Details
Note: The PR will be ready for live testing at https://rocketchat.github.io/EmbeddedChat/pulls/pr-1015 after approval. Contributors are requested to replace <pr_number> with the actual PR number.

### Testing Instructions:
- Basic Functionality: Type :sm in the message input to see emoji suggestions
- Keyboard Navigation: Use ArrowUp/Down to navigate, Enter to select, Escape to cancel
- Click Selection: Click on any emoji in the list to select it
- Integration Tests: Verify no interference with @ mentions or / commands
- Edge Cases: Test with empty input, special characters, and rapid typing

### CODE REVIEW

**Comment by Spiral-Memory** [comment]
Hi @snehalsaurabh 
Please format it with prettier and push it again. Thanks

**Comment by snehalsaurabh** [comment]
@Spiral-Memory I have done the formatting with prettier. Waiting for your review and approval.

**Comment by snehalsaurabh** [comment]
Hi @Spiral-Memory. Sorry to bother. Looking for a follow up on this and if anything else that needs to be done in here.

**Review by Spiral-Memory — APPROVED** [review]
LGTM. Approving to test. 
Thanks for the great work @snehalsaurabh

File: packages/react/src/views/EmojiList/EmojiList.js [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+      });
+    }
+  }, [emojiIndex]);
+
+  if (!filteredEmojis || filteredEmojis.length === 0) {
+    return null;
+  }
+
+  return (
+    <Box css={styles.main}>
```
Comment (Spiral-Memory) [inline packages/react/src/views/EmojiList/EmojiList.js:124]: Is it not possible to use the existing SingleSelect we have for mentions?

Comment (Spiral-Memory) [reply packages/react/src/views/EmojiList/EmojiList.js:124]: What do you think which will be better to go with ? A new custom component or SingleSelect we have

**Comment by Spiral-Memory** [comment]
This is a great PR, @snehalsaurabh. Thanks a lot for the wonderful work. Once we’re aligned on the approach, it will be merged.

**Comment by Spiral-Memory** [comment]
This PR is being merged as it’s excellent work; we anticipate other community members will transition to the existing SelectBox, including @Pratheek555 and @Khizarshah01.
